Unity-Technologies/Graphics
# [HDRP] Clarifying ProjectSettings/UserSettings Editor/Runtime assembly and proxy usage

### Purpose of this PR
Recent PR have moved HDProjectSettings to Runtime API to use the folder path here.
We splited it to have a part in HDUserSettings.
Finally, we needed it back in Editor and introduce a proxy as we wanted to add enum type in it that are defined only in Editor (#3988).
The state is now mixed, one part in Editor, the other in Runtime. And the introduced proxy broke the Project/User data separation.

This PR is trying to solve the above.
Now both HDProjectSettings and HDUserSettings are in Editor assembly while the proxy is the only one in Runtime assembly.
When trying to access the folder in Runtime assembly, we now directly use the proxy.
Project/User data separation is restored as the proxy is removed from HDUserProject.

Also, instance usage in HDProjectSettings have been cleaned, ensuring we will create/load what needed when calling recently added methods.

Serialization have not changed.

---
### Testing status
Manually tested:
- opened the wizard and check that we still can toggle without error the Show at Start.

---
### Comments to reviewers
